interfaces/sourcegen/sourcegen/headers/generator.py

```diff
@@ -0,0 +1,339 @@
+"""Resolve recipes based on specification file contents."""
+
+# This file is part of Cantera. See License.txt in the top-level directory or
+# at https://cantera.org/license.txt for license and copyright information.
+
+import sys
+import logging
+from dataclasses import dataclass
+
+from jinja2 import Environment, BaseLoader
+
+from .tagfiles import TagFileParser
+from .._dataclasses import HeaderFile, Param, ArgList, CFunc, Recipe
+from .._helpers import with_unpack_iter
+
+
+_LOGGER = logging.getLogger()
+
+
+@dataclass(frozen=True)
+@with_unpack_iter
+class Config:
+    """Provides configuration info for the HeaderGenerator class"""
+
+    ret_type_crosswalk: dict[str, str]  #: Return type cross-walks
+
+    prop_type_crosswalk: dict[str, str]  #: Parameter type cross-walks
+
+
+class HeaderGenerator:
+    """Convert recipes to header file information."""
+
+    _clib_bases: list[str] = None  #: list of bases provided via YAML configurations
+
+    def __init__(self, config: dict, templates: dict, bases: list[str]) -> None:
+        self._config = Config(**config)
+        self._templates = templates
+        self._doxygen_tags = None
+        self._clib_bases = bases
+
+    def resolve_tags(self, headers_files: list[HeaderFile]) -> None:
+        """Resolve recipe information based on doxygen tags."""
+        def get_bases() -> tuple[list[str], list[str]]:
+            bases = set()
+            classes = set()
+            for headers in headers_files:
+                bases |= {headers.base}
+                for recipe in headers.recipes:
+                    classes |= set(recipe.bases)
```

Comment:
When the items being added aren't already sets, I think it would be better to use the simpler methods:
```suggestion
                bases.add(headers.base)
                for recipe in headers.recipes:
                    classes.update(recipe.bases)
```